From lrodriguez@atheros.com  Wed Dec 16 14:39:42 2009
From: Sujith <Sujith.Manoharan@atheros.com>
Date: Wed, 16 Dec 2009 11:51:41 -0500
Subject: ath9k: Fix TX hang poll routine
To: greg@kroah.com, stable@kernel.org
Cc: linux-wireless@vger.kernel.org, Sujith <Sujith.Manoharan@atheros.com>, "Luis R. Rodriguez" <lrodriguez@atheros.com>
Message-ID: <1260982305-32244-2-git-send-email-lrodriguez@atheros.com>


From: Sujith <Sujith.Manoharan@atheros.com>

This is a backport of upstream commit: 332c556633b8c5fb4e890b1783122f2315526590

When TX is hung, the chip is reset. Ensure that
the chip is awake by using the PS wrappers.


Signed-off-by: Sujith <Sujith.Manoharan@atheros.com>
Signed-off-by: Luis R. Rodriguez <lrodriguez@atheros.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/net/wireless/ath/ath9k/xmit.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/net/wireless/ath/ath9k/xmit.c
+++ b/drivers/net/wireless/ath/ath9k/xmit.c
@@ -2065,7 +2065,9 @@ static void ath_tx_complete_poll_work(st
 
 	if (needreset) {
 		DPRINTF(sc, ATH_DBG_RESET, "tx hung, resetting the chip\n");
+		ath9k_ps_wakeup(sc);
 		ath_reset(sc, false);
+		ath9k_ps_restore(sc);
 	}
 
 	ieee80211_queue_delayed_work(sc->hw, &sc->tx_complete_work,
